x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ssl/StoreKeyConfig.java

```diff
@@ -61,7 +61,10 @@
      */
     StoreKeyConfig(String keyStorePath, String keyStoreType, SecureString keyStorePassword, SecureString keyPassword,
                    String keyStoreAlgorithm, String trustStoreAlgorithm) {
-        this.keyStorePath = keyStorePath;
+        if (keyStoreType.equals("PKCS11")) {
```

Comment:
we did specify PKCS11 in our docs, but pkcs11 would have worked too ( and other types are case insensitive too AFAIR ) so we could do `equalsIgnoreCase` here